### DISCUSSION TITLE
Is there a way to track only the first man seen

### DISCUSSION THREAD

**Discussion by jbk3407** [ultralytics/ultralytics#10224] [Q&A]
Hello ultralytics team! I'm really having fun playing with yolov8! Thanks for you hard work to make this all possible!

Anyway, I would like to track the first person seen on my webcam, and track "ONLY" that person. Would there be a way to do this?

I've thought about using the ID. But if I  do so, I'm worried about id changing issue(due to occlusion and other).
So, I thought..if I make it track only the first person seen on video(in my case webcam) this would solve changing id issue.

Could you give me some help to make this possible?

Thanks in advance :)

**Comment by pderrenger** [comment]
Hello! 🌟 We're thrilled to hear you're enjoying working with YOLOv8, and we appreciate your kind words!

To achieve your goal of tracking only the first person seen by your webcam, you can indeed utilize the ID provided by the tracking algorithm, but with a slight twist. By maintaining a tracker for the first person seen, you can just ignore new IDs after the first one is set. Here's a basic approach:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Initialize variable to store the ID of the first person
first_person_id = None

# Assume 'frame' is a frame from your webcam
results = model.track(frame)

# Check if we've already seen a person
if first_person_id is not None:
    # Filter results to only include the first person
    results = [r for r in results if r.boxes.id[0] == first_person_id]
else:
    # If we see a person, store the first person's ID
    for r in results:
        if r.boxes.cls[0] == 'person':  # Assuming 'person' class exists
            first_person_id = r.boxes.id[0]
            break

# Now 'results' either contain only the first person seen or is empty (if no person was seen yet)
```

This script starts by checking if we have already identified the first person. If not, it looks through the detections to find a person and stores their ID. Once the ID is stored, it filters future frames to only include that ID.

Remember, tracking can sometimes lose the object due to occlusions or rapid movements, so this method assumes the ID is consistently reliable once the first person is identified.

I hope this helps! Keep experimenting and feel free to reach out if you have more questions. Happy tracking! 😊
